src/typecheck/mod.rs

```diff
@@ -1381,51 +1435,88 @@ fn type_check_<L: Linearizer>(
                         )
                     })
             } else {
-                let rows = record.fields.iter().try_fold(
-                    mk_uty_row!(),
-                    |acc, (id, field)| -> Result<UnifRecordRows, TypecheckError> {
-                        // In the case of a recursive record, new types (either type variables or
-                        // annotations) have already be determined and put in the typing context,
-                        // and we need to use the same.
-                        let ty = if let Term::RecRecord(..) = t.as_ref() {
-                            ctxt.type_env.get(id).cloned().unwrap()
-                        } else {
-                            state.table.fresh_type_uvar()
-                        };
-
-                        type_check_field(
+                // Building the type {id1 : ?a1, id2: ?a2, .., idn: ?an}
+                let mut rows_expected_type: HashMap<Ident, UnifType> = record
+                    .fields
+                    .keys()
+                    .map(|id| (*id, state.table.fresh_type_uvar()))
+                    .collect();
+
+                let rows_skeleton =
+                    rows_expected_type
+                        .iter()
+                        .fold(mk_uty_row!(), |acc, (id, row_ty)| {
+                            // if let Term::RecRecord(..) = t.as_ref() {
+                            //     ctxt.type_env.get(id).cloned().unwrap()
+                            // }
```

Comment:
Is there a reason to leave this commented-out code here?

Reply:
Ah, no, it's a left-over. Thanks.